src/main/java/com/ridemart/dto/AdvertisementFilterDto.java

```diff
@@ -0,0 +1,25 @@
+package com.ridemart.dto;
+
+import com.ridemart.enums.EngineType;
+import com.ridemart.enums.FuelSystemType;
+import com.ridemart.enums.MotorbikeType;
+import com.ridemart.enums.Make;
+import lombok.Data;
+
+@Data
+public class AdvertisementFilterDto {
```

Comment:
Consider adding NoArgsConstructor and AllArgsConstructor in your DTOs. Then go over the Services, where you use these DTOs and see if it is not possible to use the AllArgsConstructor perhaps.